src/wasm/wasm-binary.cpp

```diff
@@ -1784,6 +1800,7 @@ void WasmBinaryBuilder::readFunctions() {
       }
     }
     currFunction = nullptr;
+    useDebugLocation = false;
```

Comment:
why is this change needed?

More generally, reading through the WasmBinaryBuilder's usage of `useDebugLocation`, I'm not sure I understand it. It seems to only be used locally in `WasmBinaryBuilder::readExpression`, but set in other locations, like the one added in this patch - are they even needed?

Reply:
data segments have expressions -- this fixes a crash caused by using get_global from such expression that tried to access debug location information.

Reply:
Is the crash related to the other changes in this PR? Or is this fix unrelated?

Reply:
I could reproduce crash with the debug information that now produced by the updated logic. I don't think we need an additional tests since I found this bug when was adjusting the test cases.

Reply:
Great, thanks!